modules/oe_subscriptions_anonymous/src/Controller/SubscriptionAnonymousController.php

```diff
@@ -96,9 +97,25 @@ public function confirmSubscriptionRequest(FlagInterface $flag, string $entity_i
       return $response;
     }
 
-    $this->subscriptionManager->subscribe($email, $flag, $entity_id);
     // The token has been used, so we need to invalidate it.
     $this->tokenManager->delete($email, $scope);
+
+    try {
+      $this->subscriptionManager->subscribe($email, $flag, $entity_id);
+    }
+    catch (RegisteredUserEmailException) {
+      // The email address is already taken.
+      // The user has already proven at this point that they own the email
+      // address.
+      // It is therefore ok to reveal that a user account with this email
+      // address is associated with a regular account.
+      $this->messenger()->addWarning(
+        '<p>' . $this->t('You have attempted to subscribe as anonymous, using an email address that is already associated with a regular account.') . '</p>'
+        . '<p>' . $this->t('If you still want to subscribe to content updates for this item, you can log in to the website, using your existing account, and then subscribe as a regular user.') . '</p>',
```

Comment:
oops, is the `<p>` displayed with the message? In that case I should fix it.
I just checked in core, I don't see any place where a `<p>` is added in a status message. So seems this is not something meant to happen.

Reply:
I think there are examples with other messages in the module.

Reply:
Actually, I realize that `$something . $this->t(..)` is a total anti-pattern in Drupal > 7 :)
`$this->t()` returns a markup object, and string concat converts it back to string, which we usually don't want.

Reply:
I see in other places we are using theme templates.
In this case I feel this would be overkill.
Any theme template we introduce becomes part of the API, and needs to be supported in future versions.

Alternatives:
- Put the `<p>` as part of the translatable string.
  In Drupal core I find 2 such places, both of them in `UserPasswordResetForm`.
- Use a wrapper `new FormattableMarkup('<p>@first_line</p><p>@second_line</p>', ...)`.
  For now this is the direction I would go with.

But now I notice a different problem.
If a page contains multiple message of same severity, when using Olivero or Claro, all of them appear in the same box, with no clear separation.

I get the idea that there is no clearly established best practice to put paragraphs or line breaks into status messages. Sad!

Reply:
In oe_whitelabel, such status messages are properly visually separated.
In the core themes they are not.
So we will go for `<br>`.

Reply:
I wanted to change the test to test the `<br>` in the message.
But I notice this would be too much trouble for now.
Let's keep the text-only assertion, and use a follow-up for test improvements.

Reply:
Thought about using existing template but indeed message only presents body, no title, adding the message with <br> seems the way to go.